Tests/BuildTests/BuildPlanTests.swift

```diff
@@ -104,7 +104,7 @@ final class BuildPlanTests: XCTestCase {
         let exe = try result.target(for: "exe").swiftTarget().compileArguments()
         XCTAssertMatch(exe, ["-swift-version", "4", "-enable-batch-mode", "-Onone", "-g", "-enable-testing", .equal(j), "-DSWIFT_PACKAGE", "-DDEBUG", "-module-cache-path", "/path/to/build/debug/ModuleCache", .anySequence])
 
-        let lib = try result.target(for: "lib").swiftTarget().compileArguments()
+        let lib = try result.target(for: "lib").swiftTarget().compileArguments().filter { $0 != "-application-extension" }
```

Comment:
Eh, don't special case this flag like this.

Reply:
Well, seems like it is making these tests needlessly complex which are only checking for specific flags if we’d introduce the check for Darwin etc.  ¯\\\_(ツ)\_/¯

Reply:
This will probably be a common theme going forward so it's probably worth breaking this into separate checks for the platform (e.g., a flag I added today also differed per-platform).

Reply:
Best theme.